models/models.go

```diff
@@ -8,28 +8,27 @@ import (
 type FetchMeta struct {
 	ID uint `gorm:"primary_key"`
 
-	FileName  string
+	FileName  string `gorm:"type:varchar(255)"`
 	Timestamp time.Time
 }
 
 // Root is root struct
 type Root struct {
-	ID uint `gorm:"primary_key"`
-	//  Timestamp   time.Time
-	Family      string
-	OSVersion   string
+	ID          uint   `gorm:"primary_key"`
+	Family      string `gorm:"type:varchar(255)"`
+	OSVersion   string `gorm:"type:varchar(255)"`
 	Definitions []Definition
 	Timestamp   time.Time
 }
 
 // Definition : >definitions>definition
 type Definition struct {
-	ID     uint `gorm:"primary_key"`
-	RootID uint `json:"-" xml:"-"`
+	ID     uint `gorm:"primary_key" json:"-"`
+	RootID uint `gorm:"index:idx_definition_root_id" json:"-" xml:"-"`
 
-	DefinitionID  string
+	DefinitionID  string `gorm:"type:varchar(255)"`
 	Title         string `gorm:"type:text"`
-	Description   string `gorm:"size:16777215"`
+	Description   string
```

Comment:
Does this line need a gorm struct tag?

Reply:
>  // add default size for string fields, by default, will use db type `longtext` for fields without size, not a primary key, no index defined and don't have default values
Refs. https://github.com/go-gorm/mysql/tree/15e2cbc6fd072be99215a82292e025dab25e2e16

MySQL was a problem, but GORMv2 basically uses longtext, so there is no need to specify size.

Reply:
@MaineK00n 
In the future we will think the same as we do now, so please note it in the source comments.